pandas/io/parsers.py

```diff
@@ -1474,6 +1474,13 @@ def _set(x):
                 else:
                     _set(val)
 
+        elif self.parse_dates:
+            if isinstance(self.index_col, list):
```

Comment:
`index_col` can never be a `dict` per the <a href="http://pandas.pydata.org/pandas-docs/version/0.18.1/generated/pandas.read_csv.html">docs</a>.

If you're referring to `parse_dates`, `parse_dates` being a `dict` has a completely different meaning that is independent of the `index_col`.